### DISCUSSION TITLE
[Kavita] Idea: Expand the filter/smart filter system (more exposed tags)

### DISCUSSION THREAD

**Discussion by Cantello** [Kareadita/Kavita#2741] [Ideas]
### Idea Description

Currently I have several libraries, one per language and one for magazines & newspapers (which I do not use often). For better organization, I like to create filters to have a better visibility of certain items, such as limited series (2-10 issues, mostly franco-belge comics) or graphic novels. The files are cbz and tagged with ComicRack (now that there is an updated version available), which also tags the book's format (one-shot, graphic novel, limited, etc.) and the total ~~page~~ issue count (`Format` and `Count` as per ComicInfo 1.0). 

With the current system I can only simulate this with manual tags from ComicRack (tag every graphic novel with e.g. `gn` but having this inbuilt would be much more flexible.

It would be great if Kavita can expand the tags that are exposed to the smart filter system by at least these variables

### Idea Category

Feature Enhancement

### Duration of Using Kavita

3 months

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
Btw Count is not for page count. To my knowledge, all reading software uses it for the number of issues in the series and is derived for publication status.

**Reply by Cantello** [reply]
You are completely right, I meant total issue count above, page count would be `PageCount` as far as I know.

**Comment by DieselTech** [comment]
Now that there is a comic library rewrite on the way that closely follows the comicinfo.xml metadata, it would be nice to see more of the tags used in comicinfo searchable.

Ones that are missing right now are: Teams, Location, Volume, Page Count

**Reply by majora2007** [reply]
I ended up implementing Teams and Locations as part of the Comic Rework.

**Comment by DieselTech** [comment]
Suggestion from discord:

Sort by Chapter / Volume count. That way you can have results ordered by how many chapters / volumes there are. Or to show series that have `x` number of issues.

**Comment by Cantello** [comment]
Any chance some more tags/metadata fields will be exposed to the search function?

**Reply by majora2007** [reply]
This is not a good comment. You are just saying "Can you improve" and not giving me any help in what you want. Please be explicit so I can understand other needs from the community.

**Reply by Cantello** [reply]
As I wrote above, for me `Format` and `Count` would be good tags to be exposed. Would specific use cases be helpful?

**Reply by majora2007** [reply]
Oh sorry, I misunderstood as a user in discord just asked about enhancing and I told to leave a comment, then saw this what looks like a bump request. 

But yes, more data is good. More examples and understanding on what people want from the filtering helps greatly (and helps justify the time to put in the work).

**Reply by Cantello** [reply]
So, here is my use case: I have some comics collected and while I do read ongoing comics, I currently prefer to read graphic novels, one-shots or limited series. However, with new issues of ongoing comics being released every month/week/whatever, these limited issues are quickly buried in the newly-added section and currently I cannot filter them as a smart list as the tag is not exposed to the filters. I can add tags in Comicrack (with "GN", "limited", etc.) but that is cumbersome.
So, I'd be grateful for "Format" (and with a lower priority "Count" to somewhow derive publication status, i.e. "ongoing" or "completed") to sort and read my library.

**Comment by JJMcNugget** [comment]
Two examples:

1) I would like to quickly see only those series that have a maximum of 5 volumes.
2. i would like to display the series that still have 10 volumes to read.

This is currently not possible using the Smart Filter function.

My wish would therefore be to 

1. add the option “Volume” to the Smart Filter with the selections (a) equal to, (b) equal to or more than, (c) less than or (d) equal to or less than as well as for
2. to modify the existing Smart Filter “Read Progress” so that you can switch between the options percentage and volume - currently only percentage is possible or, if this is not possible, to add a new filter “Read Progress (Volume)”.
